Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/Game.java

```diff
@@ -0,0 +1,115 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.util.ArrayList;
+
+public class Game {
+	protected 	Dictionary 	dictionary;		
+	protected 	Player 		player;		  	
+	protected 	char[] 		etalon;			
```

Comment:
I get what you're saying. I think with security I disagree because the the superclass won't know what subclasses do with them but subclasses also won't know if you ever alter the behaviour of the superclass.

In the end I would argue that the best solution is to pass as many of these in as dependencies in the constructor (I like dependency injection in case you haven't guessed :P) and make them final. Use getters if necessary, but make everything as immutable as possible. Setters to be only used if there is no choice, and exposing them as protected fields is to be avoided.